// register-js/Gruntfile.js
var fs = require("fs");
fs.writeFileSync("package.json", JSON.stringify(require("./package")));

module.exports = function(grunt) {

    grunt.initConfig({
        pkg: grunt.file.readJSON('package.json'),
        uglify: {
            options: {
                banner: '/*! <%= pkg.name %> Created at <%= grunt.template.today("dd-mm-yyyy") %> */\n',
                sourceMap: function(path) {
                    return path + ".map";
                }
            },
            dist: {
                files: {
                    'dist/register.min.js': ['register.js']
                }
            }
        },
        jshint: {
            files: ['register.js'],
            options: {
                // options here to override JSHint defaults
                globals: {
                    console: true,
                    module: true,
                    document: true,
                    window: true
                }
            }
        },
        jsdoc : {
            dist : {
                src: ['register.js'],
                options: {
                    destination: 'doc'
                }
            }
        },
        watch: {
            scripts: {
                files: ['register.js'],
                tasks: ['jsdoc'],
                options: {
                    spawn: false
                }
            }
        }
    });

    grunt.loadNpmTasks('grunt-contrib-uglify');
    grunt.loadNpmTasks('grunt-contrib-jshint');
    grunt.loadNpmTasks('grunt-contrib-watch');
    grunt.loadNpmTasks('grunt-jsdoc');

    grunt.registerTask('default', ['test', 'jshint', 'jsdoc', 'uglify']);

    //run test first;
    grunt.registerTask('test', function (){
       require("./test");
    });
};
